src/internal/db/engine.py

```
@@ -30,29 +39,29 @@ def create_engine_from_env():
 
-def get_session() -> Session:
-    """Create a Session bound to the lazily initialized Engine."""
-    global _session_factory
-    if _session_factory is None:
-        _session_factory = sessionmaker(bind=get_engine(), autoflush=False, autocommit=False)
-    return _session_factory()
+def get_session():
+    """Create a sync Session bound to the lazily initialized Engine."""
+    global _sync_session_factory
+    if _sync_session_factory is None:
+        _sync_session_factory = sessionmaker(bind=get_engine(), autoflush=False, autocommit=False)
```

Pylint (4.0.5)

[error] Undefined variable 'sessionmaker'

(E0602)